core/src/main/java/net/opentsdb/query/processor/downsample/DownsampleFactory.java

```diff
@@ -257,29 +262,29 @@ public void setupGraph(final QueryPipelineContext context,
       }
       return;
     }
-    
-    Builder builder = DownsampleConfig.newBuilder();
-    DownsampleConfig.cloneBuilder(config, builder);
-    DownsampleConfig newConfig = builder
-        .setStart(context.query().getStart())
-        .setEnd(context.query().getEnd())
-        .setId(config.getId())
-        .setResultIds(((DefaultQueryPlanner) plan).compileResultIds(config))
-        .build();
 
-    // and we need to find our sources if we have a rollup as well as set the 
-    // padding.
     final List<QueryNodeConfig> sources = Lists.newArrayList(
-        plan.terminalSourceNodes(config));
+            plan.terminalSourceNodes(config));
     for (final QueryNodeConfig source : sources) {
       if (!(source instanceof TimeSeriesDataSourceConfig)) {
         continue;
       }
-      
-      if (((TimeSeriesDataSourceConfig) source).getSummaryAggregations() == null ||
-          ((TimeSeriesDataSourceConfig) source).getSummaryAggregations().isEmpty()) {
-        final TimeSeriesDataSourceConfig.Builder new_source = 
-            (TimeSeriesDataSourceConfig.Builder) source.toBuilder();
+
+      final TimeSeriesDataSourceConfig tsConfig =
+              (TimeSeriesDataSourceConfig) source;
+      Builder builder = DownsampleConfig.newBuilder();
+      DownsampleConfig.cloneBuilder(config, builder);
+      DownsampleConfig newConfig = builder
+              .setStart(Long.toString(tsConfig.startTimestamp().msEpoch()))
+              .setEnd(Long.toString(tsConfig.endTimestamp().msEpoch()))
+              .setId(config.getId())
+              .setResultIds(((DefaultQueryPlanner) plan).compileResultIds(config))
+              .build();
+
+      if (tsConfig.getSummaryAggregations() == null ||
+          tsConfig.getSummaryAggregations().isEmpty()) {
+        final TimeSeriesDataSourceConfig.Builder new_source =
+                (TimeSeriesDataSourceConfig.Builder) source.toBuilder();
         new_source.setSummaryInterval(newConfig.getInterval());
```

Comment:
Seems to me that newConfig.getInterval() is always going to be the default interval, given that we are not giving one in the builder.

Reply:
It actually computes it in the ctor (if it's auto, etc) which is why I'm leaving it for now. We can work out a better method to call a little later.